gmso/formats/lammpsdata.py

```diff
@@ -183,48 +183,159 @@ def write_lammpsdata(topology, filename, atom_style="full"):
+                                .in_units(u.Unit("angstrom"))
+                                .value,
+                            )
+                        )
+                    elif (
+                        scaling_factor := simplify(bond_expression)
+                        / simplify(bond_type.expression)
+                    ).is_real:
+                        data.write(
+                            "{}\t{:.5f}\t{:.5f}\n".format(
+                                idx + 1,
+                                bond_type.parameters["k"]
+                                .in_units(u.Unit("kcal/mol/angstrom**2"))
+                                .value
+                                / scaling_factor,
+                                bond_type.parameters["r_eq"]
+                                .in_units(u.Unit("angstrom"))
+                                .value,
+                            )
+                        )
+                    else:
+                        raise ValueError(
+                            'Bond Style "{}" is invalid or is not currently supported'.format(
+                                bond_type.expression
+                            )
+                        )
+
+            if topology.angles:
+                data.write("\nAngle Coeffs\n\n")
+                for idx, angle_type in enumerate(topology.angle_types):
+                    # expected lammps harmonic angle expression
+                    angle_expression = "k * (theta - theta_eq)**2"
+
+                    if sympify(angle_type.expression).equals(
+                        sympify(angle_expression)
+                    ):
+                        data.write(
+                            "{}\t{:.5f}\t{:.5f}\n".format(
+                                idx + 1,
+                                angle_type.parameters["k"]
+                                .in_units(u.Unit("kcal/mol/radian**2"))
+                                .value,
+                                angle_type.parameters["theta_eq"]
+                                .in_units(u.Unit("degree"))
+                                .value,
+                            )
+                        )
+                    elif sympify(angle_type.expression).equals(
+                        sympify("0.5 * k * (theta - theta_eq)**2")
+                    ):
+                        data.write(
+                            "{}\t{:.5f}\t{:.5f}\n".format(
+                                idx + 1,
+                                angle_type.parameters["k"]
+                                .in_units(u.Unit("kcal/mol/radian**2"))
+                                .value
+                                / 2.0,
+                                angle_type.parameters["theta_eq"]
+                                .in_units(u.Unit("degree"))
+                                .value,
+                            )
+                        )
+                    elif (
```

Comment:
Similar to the comments above, the third `elif` should have covered the second `elif`

Reply:
same as above